sphinx/tutorial/cylc/furthertopics/retries.rst

```diff
@@ -0,0 +1,161 @@
+.. include:: ../../../hyperlinks.rst
+   :start-line: 1
+
+Retries
+=======
+
+Retries allow use to automatically re-submit tasks which have failed due to
+failure in submission or execution.
+
+
+Purpose 
+-------
+
+Retries can be useful for tasks that may occasionally fail due to external
+events, and are routinely fixable when they do - an example would be a task
+that is dependent on a system that experiences temporary outages.
+
+If a task fails, the cylc retry mechanism can resubmit it after a
+pre-determined delay. An environment variable, ``$CYLC_TASK_TRY_NUMBER``
+is incremented and passed into the task - this means you can write your
+task script so that it changes behaviour accordingly.
+
+
+Example
+-------
+
+.. image:: https://upload.wikimedia.org/wikipedia/commons/7/73/Double-six-dice.jpg
+   :width: 200px
+   :align: right
+   :alt: Two dice both showing the number six
+
+Create a new suite by running the following commands::
+
+   rose tutorial retries-tutorial
+   cd retries-tutorial
+
+You will now have a suite with a ``roll_doubles`` task which simulates
+trying to roll doubles using two dice:
+
+.. code-block:: cylc
+
+   [cylc]
+       UTC mode = True # Ignore DST
+
+   [scheduling]
+       [[dependencies]]
+           graph = start => roll_doubles => win
+
+   [runtime]
+       [[start]]
+       [[win]]
+       [[roll_doubles]]
+           script = """
+   sleep 10
+   RANDOM=$$  # Seed $RANDOM
+   DIE_1=$((RANDOM%6 + 1))
+   DIE_2=$((RANDOM%6 + 1))
+   echo "Rolled $DIE_1 and $DIE_2..."
+   if (($DIE_1 == $DIE_2)); then
+      echo "doubles!"
+   else
+      exit 1
+   fi
+   """
+
+
+Running Without Retries
+-----------------------
+
+Let's see what happens when we run the suite as it is. Open the cylc gui::
+
+   cylc gui retries-tutorial &
+
+Then run the suite::
+
+   cylc run retries-tutorial
+
+Unless you're lucky, the suite should fail at the roll_doubles task.
+
+Stop the suite::
+
+   cylc stop retries-tutorial
+
+
+Configuring Retries
+-------------------
+
+We need to tell cylc to retry it a few times - replace the line
+``[[roll_doubles]]`` in the ``suite.rc`` file with:
+
+.. code-block:: cylc
+
+   [[roll_doubles]]
+       [[[job]]]
+           execution retry delays = 5*PT6S
+
+This means that if the ``roll_doubles`` task fails, cylc expects to
+retry running it 5 times before finally failing. Each retry will have
+a delay of 6 seconds.
+
+We can multiple retry periods with the ``execution retry delays`` setting by
+separating them with commas, for example the following line would tell cylc to
+retry a task, first after 15 seconds, then after 10 minutes, then after one hour
+then after three hours.
+
+.. code-block:: cylc
+
+   execution retry delays = PT15S, PT10M, PT1H, PT3H
+
+We've chosen 6 seconds because it's relatively easy to observe for this example.
```

Comment:
I advise moving this line to the paragraph ending line 99 (or else omitting it as it is not strictly necessary) as it applies to the practical example discussed last there and not the above example.